services/src/main/java/org/keycloak/services/resources/admin/UsersResource.java

```diff
@@ -105,7 +100,21 @@ public UsersResource(RealmModel realm, AdminPermissionEvaluator auth, AdminEvent
     @POST
     @Consumes(MediaType.APPLICATION_JSON)
     public Response createUser(final UserRepresentation rep) {
-        auth.users().requireManage();
+        // if groups is part of the user rep, check if admin has manage_members and manage_membership on each group
+        if (rep.getGroups() != null) {
+            for (String groupId : rep.getGroups()) {
+                GroupModel group = realm.getGroupById(groupId);
+                if (group != null) {
+                    auth.groups().requireManageMembers(group);
+                    auth.groups().requireManageMembership(group);
```

Comment:
This is strange, what's the difference between requireManageMembers and requireManageMembership?

Reply:
My understanding is the following:
- MANAGE_MEMBERS_SCOPE: allowing an admin to manage users, i.e. create, update and delete users
- MANAGE_MEMBERSHIP_SCOPE: allowing an admin to manage groups membership for a specific user

So, when creating a new user part of one or more groups, we need both permissions IMO. Let me know your thoughts.

Reply:
Yeah, it makes sense. 

The `requireManageMembers` works as an alternative to `users().requireManage()` so that admins can manage users from a specific group. You are introducing (and probably fixing if we don't have an issue yet) the possibility to also create users when the admin has `MANAGE_MEMBERS_SCOPE` for a group. By doing this, you should also check the permissions when updating and deleting users. As well as provide tests to cover the behavior.

As a suggestion, I would not introduce this behavior but just create the user and assign it to a group. You could keep the `MANAGE_MEMBERSHIP_SCOPE` though.